exercicios/para-casa/entregas/entrega-elis-coimbra/projeto/Conta/Conta.js

```diff
@@ -0,0 +1,176 @@
+class Conta{
+    #agencia;
+    #conta;
+    #saldo;
+    chavesPix;
+    renda ;
+
+    static listaContas = [];
+    constructor(agencia, conta, saldo, renda){
+        this.#agencia = agencia;
+        this.#conta = conta;
+        this.#saldo = saldo;
+        this.renda = renda;
+        this.chavesPix = {
+            cpf: undefined,
+            email: undefined,
+            telefone: undefined
+        };
+
+        Conta.listaContas.push(this)
+    }
+
+    //metodo para destruir objeto da lista de contas pra salvar sua memoria
+    destruir(){
+        let i = Conta.listaContas.indexOf(this);
+        Conta.listaContas.splice(i, 1)
+    }
+
+    criarConta(agencia, conta, saldo){
+        if(agencia.length === 4 && conta.length === 5 && saldo > 0){
+            this.#agencia = agencia;
+            this.#conta = conta;
+            this.#saldo = saldo;
+        
+            return "Conta criada com sucesso";
+        } else {
+             throw new Error("Dados inválidos para cadastro");
+        }
+
+    }
+    sacar(valor){
```

Comment:
Esse método ficou com mais ifs e até um dentro do outro  o que acaba afetado a legibilidade do código, acho que seria legal algo assim:

if (typeof valor !== "number" || valor <= 0) {
    throw new Error("Valor inválido para saque");
}

if (this.#saldo - valor < 0) {
    throw new Error("Saldo insuficiente para realizar o saque");
}

const saldoAtualizado = this.#saldo - valor;
this.setSaldo(saldoAtualizado);

fazer primeiro os casos de erro e depois realizar a operação de saque de fato